claripy/ast/base.py

```diff
@@ -716,6 +719,9 @@ def replace_dict(self, replacements):
         #       raise ClaripyOperationError('cannot replace type %s ast with type %s ast' % (type(old), type(new)))
         #   old._check_replaceability(new)
 
+        if not self.variables:
```

Comment:
This changes the old behavior -- it used to be that you could replace `1` with `2` or whatnot using the ReplacementSolver, whereas with this change, that will no longer be possible. I'm not against this; just something that we need to keep in mind.